torch/_inductor/lowering.py

```diff
@@ -1035,17 +1035,14 @@ def inner_fn(idx):
                 ),
             )
 
-        def get_masked_val(i):
-            if i != len(inputs) - 1:
-                return ops.where(
-                    masks[i],
-                    masked_loads[i],
-                    get_masked_val(i + 1),
-                )
-            else:
-                return masked_loads[-1]
-
-        return get_masked_val(0)
+        next_val = masked_loads[-1]
+        for i in range((len(inputs)) - 1, -1, -1):
```

Comment:
This can be optimized a little bit by using:
```
for i in range((len(inputs)) - 2, -1, -1):
```
?

although I like binary search more but that may be not trivial for codegen..